### ISSUE TITLE
[ISSUE] Tables API does not parse table_constraints

### ISSUE THREAD

**Issue by themattmorris** [databricks/databricks-sdk-py#435]
**Description**
`TableInfo.table_constraints` is always empty when using `WorkspaceClient.tables.get()`, regardless of whether constraints exist in that table or not. The json returned by the API call includes these constraints, but they are not correctly parsed by `TableInfo.from_dict`.

**Reproduction**

```python
from databricks.sdk import WorkspaceClient


w = WorkspaceClient()
my_table = "catalog.schema.book"  # A table with primary and/or foreign key constraints
w.tables.get(my_table).table_constraints
```

```
TableConstraintList(table_constraints=None)
```

**Expected behavior**

In this case, the result from the api call returns:

```python
{
# other attributes...
 'table_constraints': [{'primary_key_constraint': {'child_columns': ['id'],
                                                   'name': 'book_pk'}},
                       {'foreign_key_constraint': {'child_columns': ['publisher_id'],
                                                   'name': 'publisher_fk',
                                                   'parent_columns': ['id'],
                                                   'parent_table': 'catalog.schema.publisher'}}],
}
```

So I expect the constraints to look like:

```python
TableConstraintList(
    table_constraints=[
        TableConstraint(
            primary_key_constraint=PrimaryKeyConstraint(
                name="book_pk",
                child_columns=["id"],
            ),
            foreign_key_constraint=ForeignKeyConstraint(
                name="publisher_fk",
                child_columns=["publisher_id"],
                parent_table="catalog.schema.publisher",
                parent_columns=["id"],
            ),
        ),
    ],
)
```

**Is it a regression?**
No

**Debug Logs**
The SDK logs helpful debugging information when debug logging is enabled. Set the log level to debug by adding `logging.basicConfig(level=logging.DEBUG)` to your program, and include the logs here.

```
DEBUG:urllib3.connectionpool:Resetting dropped connection: adb-****************.*.azuredatabricks.net
DEBUG:urllib3.connectionpool:https://adb-****************.*.azuredatabricks.net:443 "GET /api/2.1/unity-catalog/tables/catalog.schema.book HTTP/1.1" 200 None
DEBUG:databricks.sdk:GET /api/2.1/unity-catalog/tables/catalog.schema.book
< 200 OK
```

**Other Information**
 - OS: linux (rhel)
 - Version: 0.12.0

**Additional context**
If you're open to it I can submit a PR to address this, as I'd really like to be able to use this. Currently having to patch it myself anyways.

**Comment by themattmorris** [comment]
FWIW the problem is that "table_constraints" key is being called twice, both [here](https://github.com/databricks/databricks-sdk-py/blob/a862adc409e0be24f4c207737b5d5055a60d317d/databricks/sdk/service/catalog.py#L2414) and [here](https://github.com/databricks/databricks-sdk-py/blob/a862adc409e0be24f4c207737b5d5055a60d317d/databricks/sdk/service/catalog.py#L2535). This does follow what is in the [Databricks REST API reference](https://docs.databricks.com/api/workspace/tables/get):

![image](https://github.com/databricks/databricks-sdk-py/assets/16111337/7387c30e-b76d-4ea4-b81b-4c1ed4746a2c)

But the json returned by the API call does not match this structure, as the "table_constraints" key is not nested twice like this. I'm not sure if the API reference is just wrong or where the issue lies, but there is a mismatch between the structure in the API reference and what is actually returned by the API call.

If is confusing enough that there is a `TableConstraintList` class that only holds a single attribute that is a `list[TableConstraint]`, where simply having `table_constraints` be an `Optional[list[TableConstraint]]` would be easier to follow, but I understand this design decision was done on the Databricks API.

**Comment by meretri** [comment]
I have the same issue. Any info if this will be fixed some time soon?

**Comment by themattmorris** [comment]
> I have the same issue. Any info if this will be fixed some time soon?

FWIW, this is the workaround I am doing in the meantime if it's of any help to you:

```python
from databricks.sdk.service.catalog import TableInfo


data = w.api_client.do(
    "GET",
    "/api/2.1/unity-catalog/tables/<TABLE_NAME>", 
    query={"include_delta_metadata": True},
    headers={"Accept": "application/json"},
)
constraint_key = "table_constraints"
data[constraint_key] = {constraint_key: data.pop(constraint_key, None)}
info = TableInfo.from_dict(data)
```

**Comment by meretri** [comment]
@themattmorris thanks a lot!

**Comment by themattmorris** [comment]
Closing this issue as this was fixed in the latest release (0.17.0):

https://github.com/databricks/databricks-sdk-py/blob/21ad5661507edab0f71012ed6ae0bf127ae1c055/databricks/sdk/service/catalog.py#L3463

I have run this myself and can confirm that the table constraints are indeed being obtained. It looks like this was fixed in the databricks API:

<img width="260" alt="image" src="https://github.com/databricks/databricks-sdk-py/assets/16111337/d9db5f37-a243-401e-85f8-3eeb91aa0878">

fyi @meretri
